File: app/src/main/java/in/kashewdevelopers/passwordgenerator/SavedAdapter.java
package in.kashewdevelopers.passwordgenerator;

import android.content.Context;
import android.database.Cursor;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CursorAdapter;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

public class SavedAdapter extends CursorAdapter {

    private LayoutInflater cursorInflator;

    SavedAdapter(Context context, Cursor cursor, int flags) {
        super(context, cursor, flags);
        cursorInflator = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }


    @Override
    public View newView(Context context, Cursor cursor, ViewGroup viewGroup) {
        return cursorInflator.inflate(R.layout.saved_item_layout, null);
    }


    @Override
    public void bindView(View view, final Context context, Cursor cursor) {
        // initialize widgets
        LinearLayout parent = view.findViewById(R.id.parent);
        TextView passwordTV = view.findViewById(R.id.password);
        ImageView deleteIcon = view.findViewById(R.id.delete_icon);

        // get data from cursor
        final String password = cursor.getString(cursor.getColumnIndex(SavedDbHelper.PSWD));

        passwordTV.setText(password);

        parent.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (onPasswordClickListener != null) {
                    onPasswordClickListener.onPasswordClickListener(password);
                }
            }
        });

        deleteIcon.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (onDeleteClickListener != null) {
                    onDeleteClickListener.onDeleteClickListener(password);
                }
            }
        });
    }


    void setOnPasswordClickListener(OnPasswordClickListener onPasswordClickListener) {
        this.onPasswordClickListener = onPasswordClickListener;
    }

    void setOnDeleteClickListener(OnDeleteClickListener onDeleteClickListener) {
        this.onDeleteClickListener = onDeleteClickListener;
    }

    interface OnDeleteClickListener {
        void onDeleteClickListener(String password);
    }

    interface OnPasswordClickListener {
        void onPasswordClickListener(String password);
    }

    private OnPasswordClickListener onPasswordClickListener;
    private OnDeleteClickListener onDeleteClickListener;

}

File: app/src/main/java/in/kashewdevelopers/passwordgenerator/SavedAdapter.kt
package `in`.kashewdevelopers.passwordgenerator

import `in`.kashewdevelopers.passwordgenerator.databinding.SavedItemLayoutBinding
import android.content.Context
import android.database.Cursor
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CursorAdapter

class SavedAdapter(context: Context, cursor: Cursor) : CursorAdapter(context, cursor, 0) {

    private var cursorInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

    override fun newView(context: Context?, cursor: Cursor?, parent: ViewGroup?): View {
        return cursorInflater.inflate(R.layout.saved_item_layout, null)
    }

    override fun bindView(view: View?, context: Context?, cursor: Cursor?) {
        val binding: SavedItemLayoutBinding = SavedItemLayoutBinding.bind(view as View)

        val password = cursor?.getString(cursor.getColumnIndex(SavedDbHelper.PSWD))
        binding.password.text = password

        binding.parent.setOnClickListener {
            onPasswordClickListener?.onPasswordClickListener(password ?: "")
        }

        binding.deleteIcon.setOnClickListener {
            onDeleteClickListener?.onDeleteClickListener(password ?: "")
        }
    }


    fun setOnDeleteClickListener(onDeleteClickListener: OnDeleteClickListener) {
        this.onDeleteClickListener = onDeleteClickListener
    }

    fun setOnPasswordClickListener(onPasswordClickListener: OnPasswordClickListener) {
        this.onPasswordClickListener = onPasswordClickListener
    }

    interface OnDeleteClickListener {
        fun onDeleteClickListener(password: String)
    }

    interface OnPasswordClickListener {
        fun onPasswordClickListener(password: String)
    }

    private var onDeleteClickListener: OnDeleteClickListener? = null
    private var onPasswordClickListener: OnPasswordClickListener? = null

}